backend/src/main/java/kernel360/techpick/feature/infrastructure/tag/reader/TagReaderImpl.java

```diff
@@ -1,4 +1,39 @@
 package kernel360.techpick.feature.infrastructure.tag.reader;
 
+import java.util.List;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.stereotype.Component;
+
+import kernel360.techpick.core.model.tag.Tag;
+import kernel360.techpick.core.model.tag.TagRepository;
+import kernel360.techpick.feature.domain.tag.exception.ApiTagException;
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor
 public class TagReaderImpl implements TagReader {
+
+	private TagRepository tagRepository;
+
+	@Override
+	public Tag readTag(Long userId, Long tagId) {
```

Comment:
p3: userId가 아니라, User로 받게 되면
더 깔끔하지 않을까? id로 받다보니 getId를 한번 더 호출하게 됨
```suggestion
	ObjectUtils.notEqual(tag.getUser(), user); 
```

Reply:
아래와 같이 User 객체를 가져오기 위해서 select query가 발생함.
이를 최소화하기 위해 userId 값만 시큐리티 토큰에서 가져온 다음에 사용하였음.
```java
	@Override
	public User readCurrentUser() {
		var authentication = SecurityContextHolder.getContext().getAuthentication();
		long userId = (long)authentication.getPrincipal();
		return userRepository.findById(userId).orElseThrow(ApiUserException::USER_NOT_FOUND);
	}
```